server.hpp

```diff
@@ -160,33 +188,42 @@ class Room {
             deliver_ui_state();
             break;
         }
-
         case OpCode::READY_OP: {
-            std::cout << "ready: is_local = " << participant->is_local << ", data1 = " << data1 << ", data2 = " << data2 << std::endl;
+            logger->info("ready: is_local = {}, data1 = {}, data2 = {}", participant->is_local, data1, data2);
 
             if (contest.status == Contest::Status::GAME_OVER) {
                 contest.clear();
             }
 
             auto name { data1 };
             Role role { data2 };
-            if (!Player::is_valid_name(name))
-                name = "Player" + std::to_string(contest.players.size() + 1);
 
+            if (!Player::is_valid_name(name)) {
+                name = "Player" + std::to_string(contest.players.size() + 1);
+                logger->warn("READY_OP: invalid name {}, use default name: {}", data1, name);
+            }
             participant->set_name(name);
-
-            if (participant->is_local) {
-                Player player { participant, name, role, PlayerType::LOCAL_HUMAN_PLAYER };
-                contest.enroll(std::move(player));
-                contest.local_role = role;
-                // TODO: support multiple remote waiting players
-                deliver_to_others(msg, participant);
-            } else {
-                Player player { participant, name, role, PlayerType::REMOTE_HUMAN_PLAYER };
-                contest.enroll(std::move(player));
-                deliver_to_local(msg);
+            Player player;
+            try {
+                if (participant->is_local) {
+                    player = Player { participant, name, role, PlayerType::LOCAL_HUMAN_PLAYER };
```

Comment:
Split this huge code blick. You can first determine the player type. like 
```c++
auto player_type { participant->is_local ? PlayereType::LOCAL_HUMAN_PLAYER : PlayerType::REMOTE_HUMAN_PLAYER};
```
And construct the player.
Then, use the try-catch block and enroll the player.
Finally, deliver_to_what based on is_local